daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/SeparatedSequenceParsers.scala

```diff
@@ -35,9 +35,8 @@ trait Separated { self: SequenceChildParser =>
 sealed trait RepeatingSeparatedPoU extends Separated { self: RepeatingChildParser =>
 
   private lazy val shouldRemoveZLStringHexBinaryValue_ =
-    false && // disable this feature as many test in daffodil-core depend on NOT removing these empty strings.
+    true ||
```

Comment:
This is where the more subtle code has to go, so that we implement the DFDL spec - and have a IBM compatible mode if needed.  IBM agressively removes zero-length strings/hexBinaries. The DFDL spec was clarified to make creation of elements have priority over ZL removal if - ZL is the nil representation or ZL is the empty string or empty hexBinary representation, i.e., if there's no syntax or emptyValueDelimiterPolicy to tell it that empty string should have distinguishing syntax, so ZL should NOT become a value.